src/Product/Tests/Repositories/ProductTest.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+namespace Develop\Business\Product\Tests\Repositories;
+
+use Develop\Business\Product\Exceptions\ProductException;
+use Develop\Business\Product\Product;
+use Develop\Business\Product\Repositories\Product as ProductRepository;
+use Prophecy\Argument;
+
+class ProductTest extends \PHPUnit_Framework_TestCase
+{
+    public function testFindAllProducts()
+    {
+        $product = $this->prophesize(Product::class);
+
+        $pdo = $this->prophesize(\PDO::class);
+        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION)->shouldBeCalled();
+
+        $statement = $this->prophesize(\PDOStatement::class);
+        $statement->setFetchMode(\PDO::FETCH_CLASS | \PDO::FETCH_PROPS_LATE, Product::class, ['', 0.0, 0])->shouldBeCalled();
+        $statement->execute()->shouldBeCalled();
+        $statement->fetchAll()->willReturn([$product->reveal()]);
```

Comment:
Why not to use stubs instead mocks?
The internal calls (`setFetchMode`, `execute`, etc) really need to be validated?

Appears me that it can create a hard coupling on your tests.